src/test/java/online/book/store/controller/ShoppingCartControllerTest.java

```diff
@@ -0,0 +1,189 @@
+
+        UsernamePasswordAuthenticationToken authentication =
+                new UsernamePasswordAuthenticationToken(user, null,
+                        List.of(new SimpleGrantedAuthority("ROLE_USER")));
+        SecurityContextHolder.getContext().setAuthentication(authentication);
+    }
+
+    @WithMockUser(username = "user", roles = {"USER"})
+    @Test
+    @DisplayName("Find shopping cart by existing user - return a shopping cart")
+    void getShoppingCartByUser_ExistingShoppingCart_ReturnShoppingCart() throws Exception {
+        Long userId = 1L;
+        CartItemResponseDto expectedCartItem = new CartItemResponseDto()
+                .setId(1L)
+                .setBookId(1L)
+                .setBookTitle("Title1")
+                .setQuantity(3);
+
+        ShoppingCartDto expected = new ShoppingCartDto()
+                .setId(1L)
+                .setUserId(userId)
+                .setCartItemsDto(Set.of(expectedCartItem));
+
+        MvcResult result = mockMvc.perform(get("/cart")
+                        .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+
+        ShoppingCartDto actual = objectMapper.readValue(result.getResponse().getContentAsString(),
+                ShoppingCartDto.class);
+        assertNotNull(actual);
+        assertNotNull(actual.getId());
+        assertTrue(reflectionEquals(expected, actual, "id"));
+    }
+
+    @WithMockUser(username = "user", roles = {"USER"})
+    @Test
+    @DisplayName("Add book to shopping cart with valid request dto - success")
+    void addBookToShoppingCart_ValidRequestDto_Success() throws Exception {
+        Long userId = 1L;
+        CartItemRequestDto requestDto = new CartItemRequestDto()
+                .setBookId(2L)
+                .setQuantity(1);
+        String jsonRequest = objectMapper.writeValueAsString(requestDto);
+
+        MvcResult result = mockMvc.perform(post("/cart")
+                        .content(jsonRequest)
+                        .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+
+        ShoppingCartDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(), ShoppingCartDto.class);
+
+        Set<CartItemResponseDto> expectedCartItems = new HashSet<>();
+        expectedCartItems.add(new CartItemResponseDto()
+                .setId(1L)
+                .setBookId(1L)
+                .setBookTitle("Title1")
+                .setQuantity(3));
+        expectedCartItems.add(new CartItemResponseDto()
+                .setId(2L)
+                .setBookId(2L)
+                .setBookTitle("Title2")
+                .setQuantity(1));
+
+        ShoppingCartDto expected = new ShoppingCartDto()
+                .setId(1L)
+                .setUserId(userId)
+                .setCartItemsDto(expectedCartItems);
```

Comment:
better move it to the private method or util class. the same for other test cases